core/ops.rs

```diff
@@ -20,19 +22,84 @@ use std::ops::DerefMut;
 use std::pin::Pin;
 use std::rc::Rc;
 
-pub type OpAsyncFuture = Pin<Box<dyn Future<Output = Box<[u8]>>>>;
-pub type OpFn = dyn Fn(Rc<RefCell<OpState>>, BufVec) -> Op + 'static;
+pub use erased_serde::Serialize as Serializable;
+pub type RcOpState = Rc<RefCell<OpState>>;
+pub type PromiseId = u64;
+pub type OpBuf = Option<ZeroCopyBuf>;
```

Comment:
I don't really like this type alias either, `OpBuf` suggests that there is a buffer, I believe using `Option<ZeroCopyBuf>` will be more readable

Reply:
It's a nit but I'm with @bartlomieju on this. It's nice to have Option types be explicit. I think this type alias only makes code a bit less explicit.

I also agree with removing the `RcOpState` alias, but it's a bit more complex and thus more understandable. So if you were to remove only one of these, please remove `OpBuf`.

Reply:
> It's a nit but I'm with @bartlomieju on this. It's nice to have Option types be explicit. I think this type alias only makes code a bit less explicit.
> 
> I also agree with removing the `RcOpState` alias, but it's a bit more complex and thus more understandable. So if you were to remove only one of these, please remove `OpBuf`.

I'll undo the aliases, I think it's more important to land this and we can debate the aliases in follow-up conversations.

But for the context/record, the aliases served a few things IMO:
1. Reducing verbosity & improving consistency
2. Progressive disclosure of complexity (so unused "features" have a minimal complexity tax)
3. Given that the `OpFn` signature is controlled by one section of the codebase, but then consumed by ~200+ ops, I view it as the ops implementing and following the op interface, hence the arg types are somewhat abstract interfaces to them rather than specific concrete types
4. Ops don't really care about the specific type of `OpState`, they use it as an interface to get/set typed-state & resources
5. Nearly 90% of the ops don't use the buffer-arg and hence don't care about its type, we could add buffer-less json ops (like TSC does) but that would cause fragmentation

I get that aliasing types, especially `Option<T>`, isn't very rusty and would be a mess if done everywhere, I think there's something to consider in this case when it's ignored 90% of the time and types are more of an interface than concrete implementation types.

I'll unwrap the aliases, but I think this warrants some extra thought and consideration in the future.